develop/tutorials/articles/01-introduction-to-liferay-development/50-finding-extension-points/00-finding-extension-points-intro.markdown

```diff
@@ -0,0 +1,130 @@
+    keywords in element names and descriptions. The keyword *LDAP* resides under
+    the Liferay Foundation app suite; select it.
+
+    ![Figure x: The Liferay Foundation app suite contains the LDAP Authentication application.](../../../images/ldap-keyword-app-manager.png)
+
+3.  Select the *LDAP* application from the app listing.
+
+4.  The LDAP application only has one module, but typically, application's have
+    more than one module to inspect. Select the *Liferay Portal Security LDAP*
+    module.
+
+    ![Figure x: The App Manager lists the module, package name, version, and status.](../../../images/app-manager-breakdown.png)
+
+5.  Search through the components, again applying your keywords as a guide. Copy
+    the component name you think best fits the functionality you want to
+    customize; you'll inspect it later using the Gogo shell.
+
+    ![Figure x: The component name can be found using the App Manager.](../../../images/usermodellistener-component.png)
+
+    +$$$
+
+    **Note:** When using the Gogo shell later, understand that it can take
+    several tries to find the component for which you're looking; naming
+    conventions should allow you to find your desired extension point in a
+    manageable timeframe.
+
+    $$$
+
+Next, you'll begin using the Gogo shell to inspect the component for extension
+points.
+
+## Finding Extension Points in a Component
+
+Once you have the component that relates to the functionality you want to
+extend, you can use the Gogo shell's Service Component Runtime (SCR) commands to
+inspect it. You can execute SCR commands using
+[Liferay Blade CLI](/develop/tutorials/-/knowledge_base/7-0/blade-cli) or in
+[Gogo shell via telnet](/develop/reference/-/knowledge_base/7-0/using-the-felix-gogo-shell).
+This tutorial assumes you're using the Gogo shell via telnet.
+
+Execute the following command:
+
+    scr:info <COMPONENT_NAME>
+
+For the LDAP example component you copied previously, the command would look
+like this:
+
+    scr:info com.liferay.portal.security.ldap.internal.messaging.UserImportMessageListener
+
+The resulting SCR information includes a lot of information. For this exercise,
+you're interested in the services the component references. These are extension
+points. For example, here's the reference for the service that imports LDAP
+users:
+
+    ...
+    Reference: LdapUserImporter
+    Interface Name: com.liferay.portal.security.ldap.exportimport.LDAPUserImporter
+    Cardinality: 1..1
+    Policy: static
+    Policy option: reluctant
+    Reference Scope: bundle
+    ...
+
+The `LDAPUserImporter` is the extension point needed to customize the process of
+importing users with LDAP! If none of the references satisfy what you're looking
+for, search other components from the App Manager. If you want to learn how to
+override a component's service reference, visit the following
+[tutorial](/develop/tutorials/-/knowledge_base/7-0/overriding-a-components-service-reference).
```

Comment:
It would be good to mention the significance of a reference's policy and policy option. If the policy is `static` and the policy option is `reluctant`, binding a new higher ranking service in place of a bound service requires reactivating the component.